src/Appwrite/Utopia/Response/Model/User.php

```diff
@@ -23,6 +23,24 @@ public function __construct()
                 'default' => '',
                 'example' => 'John Doe',
             ])
+            ->addRule('password', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Hashed user password.',
+                'default' => '',
+                'example' => '$argon2id$v=19$m=2048,t=4,p=3$aUZjLnliVWRINmFNTWMudg$5S+x+7uA31xFnrHFT47yFwcJeaP0w92L/4LdgrVRXxE',
+            ])
+            ->addRule('hash', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Password hashing algorithm.',
+                'default' => '',
+                'example' => 'bcrypt',
+            ])
+            ->addRule('hashOptions', [
```

Comment:
> In the demo, the empty default was [] instead of {}.
I think this is a bug of utopia/database. Should I look at it as part of this feature?

> I think it would make more sense to have relevant hash models added
Agree, great suggestion.

> I guess we only need one for argo and modified script?
All algos except PHPass and MD5 has some options. But we usually don't need them. We only need them for SCryptModified. For the others, even tho we don't need them, they will be valuable when exporting users from Appwrite, so their new backend can use exact same options to keep exact same level of security.

I will implement Hash models. Regarding PHPass and MD5, should we have a model too? That would just be empty - no rules? IMO it would be a great indicator (in docs) that no options go there. Otherwise, people would simply not find a model for their algo and would be confused.

Reply:
> I think this is a bug of utopia/database. Should I look at it as part of this feature?

Note: Look at how we do it in preferences, should be a solution. Maybe even vars attribute in webhook/function.

Reply:
This should not be resolved due to more refactoring.

Reply:
I still have some issues with this attribute. It's very hard to describe. What kind of structure can it have?

Reply:
I took what we did for attributes, and implemented it here. Please revisit and let me know if you like it.
Commit: https://github.com/appwrite/appwrite/pull/2747/commits/df365806576467bce9292b31c963b8815d86f4d4

Reply:
Oh, right. I took the snake case from the PHP functions I use.

Updated in this PR: https://github.com/appwrite/appwrite/pull/2747/commits/0aaee54a63cd9ed18c27f8906050802980fcb158
We now use camelCase, just like all other places.